pkg/events/derive/hidden_kernel_module.go

```diff
@@ -74,94 +182,100 @@ func clearMap(bpfMap *bpf.BPFMap) error {
 				err = bpfMap.DeleteKey(unsafe.Pointer(&addr))
 
 				if err != nil {
-					logger.Errorw("DeleteKey", "error", err.Error())
+					logger.Errorw("Err occurred DeleteKey: " + err.Error())
 					return err
 				}
 			}
+			err = iter.Err()
+			if err != nil {
+				logger.Errorw("ClearMap iterator received an error", "error", err.Error())
+				return iter.Err()
+			}
 
 			return nil
 		},
+		cap.SYS_ADMIN, // Required for (older) kernels that require CAP_SYS_ADMIN to interact with eBPF maps
 	)
 
 	return err
 }
 
-func ClearModulesState(modsMap *bpf.BPFMap) {
-	_ = clearMap(modsMap)
+// GetWakeupChannelRead returns the reading-end of the channel
+func GetWakeupChannelRead() <-chan ScanRequest {
+	return wakeupChannel
 }
 
-func FillModulesFromProcFs(kernelSymbols helpers.KernelSymbolTable, modulesFromProcFs *bpf.BPFMap) error {
-	var file *os.File
-	var err error
+// ClearModulesState clears the map
+func ClearModulesState() {
+	_ = clearMap(allModsMap)
+	_ = clearMap(recentDeletedModulesMap)  // We only care for modules that got deleted in the midst of a scan (and since we're not scanning now, clear the map)
+	_ = clearMap(recentInsertedModulesMap) // We only care for modules that got inserted in the midst of a scan (and since we're not scanning now, clear the map)
+}
 
-	err = capabilities.GetInstance().Specific(
+// FillModulesFromProcFs fills a map with modules from /proc/modules, to be checked in kernel-space for inconsistencies.
+func FillModulesFromProcFs(kernelSymbols helpers.KernelSymbolTable) error {
+	err := capabilities.GetInstance().Specific(
```

Comment:
Should use the `EBPF` capabilities RING (it already contains the base line for all capabilities needed by picked events PLUS the ones needed for eBPF functions).

Reply:
IF reading /proc/modules here needs another capability, then you can either add the capability to the base line (by adding as a requirement in the events.go event) OR, if this code is not **that frequent** you can use `Specific()` here, but with the capability event needed to read /proc/ fs only, not CAP_SYS_ADMIN).